### DISCUSSION TITLE
Meteor shower local hourly rate depends on zoom level

### DISCUSSION THREAD

**Discussion by poxxo** [Stellarium/stellarium#3549] [Q&A] [answered]
The local hourly rate (LHR) displayed for the currently selected meteor shower depends on zoom level. If zoomed in such that the vertical field of view is about 50 degrees or less, LHR appears to be a reasonable estimate. However, it drops significantly upon zooming out until a vertical field of view of about 100 degrees, after which it stays constant. Is this a bug? I can't think of a reason why this would be the intended behavior. I'm using v23.3 on Windows 11.

I'm attaching two screenshots to illustrate the issue for the Geminid meteor shower at a time the radiant is close to zenith (within 5 degrees). The first image is zoomed in and shows LHR is equal to the zenithal hourly rate (ZHR). The second image is zoomed out, and LHR is much lower. The particular direction that the view is centered around has no effect (even if facing nadir).

![zoomed_in](https://github.com/Stellarium/stellarium/assets/50934161/d22e6ec3-c53b-471a-9230-5a198f084353)
![zoomed_out](https://github.com/Stellarium/stellarium/assets/50934161/3564b419-2d72-467f-a095-f9515d02c961)

**Comment by alex-w** [comment]
From the code: Limiting magnitude could be higher than 6.5, but we can't see very faint meteors, and we should set the limit to prevent unrealistic super-high meteor rate. Hourly rate that goes higher than ZHRmax may also confuse users if we allow it.

**Comment by worachate001** [comment, marked as answer]
Thank you for your observation. Limiting magnitude (LM) of the sky effects our local hourly rate calculation. Limiting magnitude goes lower in value when zoom-out (fewer stars/meteors) and higher when zoom-in (more stars/meteors visible). We can ignore the location's LM and assume that it is 6.5, but it would be misleading (too high meteor rate) for a light-polluted area.

This makes me think that it maybe better if we use only the location's LM and ignore changing LM when zooming-in or out.

**Comment by poxxo** [comment]
Thank you Alex and Worachate. I've also just come to the same realization upon looking at the code.

I personally find the current behavior somewhat counterintuitive (if anything you would expect to see more meteors per hour if you are viewing a wider portion of the sky), so the alternative suggested by Worachate makes more sense to me. At the very least it would be useful to display the limiting magnitude used in the LHR calculation. I will leave it to you guys to decide how to go about it, if you think a modification is warranted.
